src/Microsoft.ML.Data/Utilities/ColumnCursor.cs

```diff
@@ -79,6 +78,23 @@ public static IEnumerable<T> GetColumn<T>(this IDataView data, IHostEnvironment
             throw env.Except($"Could not map a data view column '{columnName}' of type {colType} to {typeof(T)}.");
         }
 
+        /// <summary>
+        /// Return a <see cref="IHost"/> assigned in a given <see cref="IDataView"/> implementation.
+        /// </summary>
+        /// <param name="data">an <see cref="IDataView"/> implementation.</param>
+        private static IHost RetrieveHost(IDataView data)
```

Comment:
>RetrieveHost [](start = 29, length = 12)

No.
Either IDataView will have IHost property/field as part of interface, or we don't do it at all.
This reflection magic will only work if you have IHost in IDataView, and there is no guarantee of that.

Reply:
Fixed. We use `Constracts` now.

---
In reply to: [260409959](https://github.com/dotnet/machinelearning/pull/2738#discussion_r260409959) [](ancestors = 260409959)